### PR TITLE
Ticket system

### PR DESCRIPTION
**Pull request by alicepotter**
## Context
In this PR, I implemented the front end and back end of the ticket system pages.

## What's New
- Added a ticket icon in the bottom bar navigation.
- `Ticket`: A data class representing a ticket.
- `MyTickets.kt`: A screen that displays the user's tickets.
- `ScanTicket.kt`: A screen that will be used to scan tickets.
- `TicketDetails.kt`: A screen showing the details of a ticket.
- `TicketViewModel` : Necessary for the backend

## Testing
- `MyTicketsScreen`
- `MyTicketsTest`
- `ScanTicketScreen`
- `ScanTicketTest`
- `TicketDetailsScreen`
- `TicketDetailsTest`

## What's to Come
The content of the scan a ticket screen will come shortly. 

## Demo

https://github.com/The-Software-Enterprise/assos/assets/91310864/acb8a1c4-2606-4dd8-9dca-997cb8598978

### CODE REVIEW

File: app/src/main/java/com/swent/assos/model/data/Ticket.kt
Comment (JulesHerrscher) [inline app/src/main/java/com/swent/assos/model/data/Ticket.kt:1]: the ticket class requires only an id and an eventid and maybe a userId.

**Review by JulesHerrscher — CHANGES_REQUESTED** [review]
nice changes to start the implementation of the tickets, but database changes are mandatory as well as tests.

File: app/src/main/java/com/swent/assos/ui/screens/ticket/TicketDetails.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+          Text(
+              text = "Start Time: " + event.startTime?.let { dateToReadableString(it) },
+              modifier = Modifier.padding(8.dp).testTag("StartTime"))
+          Text(
+              text = "End Time: " + event.endTime?.let { dateToReadableString(it) },
+              modifier = Modifier.padding(8.dp).testTag("EndTime"))
+          Text(
+              text = "Description: ${event.description}",
+              modifier = Modifier.padding(8.dp).testTag("Description"))
```
Comment (Julio12344) [inline app/src/main/java/com/swent/assos/ui/screens/ticket/TicketDetails.kt:74]: For the text use the Typography Theme.

**Review by Julio12344 — CHANGES_REQUESTED** [review]

**Review by Julio12344 — APPROVED** [review]

File: app/src/androidTest/java/com/swent/assos/MyTicketsTest.kt
Comment (JulesHerrscher) [inline app/src/androidTest/java/com/swent/assos/MyTicketsTest.kt:1]: nice to already have tests !

File: app/src/androidTest/java/com/swent/assos/ScanTicketTest.kt
Comment (JulesHerrscher) [inline app/src/androidTest/java/com/swent/assos/ScanTicketTest.kt:1]: same here, nice to have tests for the scan tickets, good practice to directly fill the datacache to "mock" the database.

File: app/src/androidTest/java/com/swent/assos/TicketDetailsTest.kt
Comment (JulesHerrscher) [inline app/src/androidTest/java/com/swent/assos/TicketDetailsTest.kt:1]: There is just some commented code.

File: app/src/main/java/com/swent/assos/model/data/Ticket.kt
Comment (JulesHerrscher) [inline app/src/main/java/com/swent/assos/model/data/Ticket.kt:1]: Nice changes !

File: app/src/main/java/com/swent/assos/model/view/TicketViewModel.kt
Comment (JulesHerrscher) [inline app/src/main/java/com/swent/assos/model/view/TicketViewModel.kt:1]: clean viewModel. Sonarcloud indicates that the dispatchers are useless when used with only suspending functions.

**Review by JulesHerrscher — APPROVED** [review]
Nice changes, it satisfies all the requirements to get merged.

**Pull request merged**